db/config.cc

```diff
@@ -972,7 +972,10 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , wasm_udf_memory_limit(this, "wasm_udf_memory_limit", value_status::Used, 2*1024*1024, "How much memory each WASM UDF can allocate at most")
     , relabel_config_file(this, "relabel_config_file", value_status::Used, "", "Optionally, read relabel config from file")
     , object_storage_config_file(this, "object_storage_config_file", value_status::Used, "", "Optionally, read object-storage endpoints config from file")
-    , minimum_keyspace_rf(this, "minimum_keyspace_rf", liveness::LiveUpdate, value_status::Used, 0, "The minimum allowed replication factor when creating or altering a keyspace.")
+    , minimum_replication_factor_fail_threshold(this, "minimum_replication_factor_fail_threshold", liveness::MustRestart, value_status::Used, -1, "")
```

Comment:
If we start doing MustRestart for guard rails, perhaps we need to go back and do it for all of them, not just this one.
I'm not sure I'm happy about this. I think @tzach is mistaken here - why is Scylla Cloud worried specifically about guardrails being changable via CQL? I would think there are much more "dangerous" options to change - people can disable hints, can turn on or off various protocols, and other strange changes, why are we worried only about guardrails? Maybe Scylla cloud needs to disable *most* configuration changes?

Reply:
I thought we agreed these can be set only in the config file, the discussion has been reopened, and we're waiting for a final decision.

Reply:
> If we start doing MustRestart for guard rails, perhaps we need to go back and do it for all of them, not just this one.

Not sure this is required, since all of them are disabled (and I don't see any our clients using them).

> I thought we agreed these can be set only in the config file, the discussion has been reopened, and we're waiting for a final decision.

Since there's no resolution, I think I can change it back to `LiveUpdate` - that shouldn't hurt any of our client, be it on premise or cloud. We can still change it later.
So can we proceed with this PR?

Reply:
LiveUpdate is not about CQL. It's about not having to restart the server.

Reply:
Right, because there's also an option to send `SIGHUP` to reload the `scylla.yaml` config.

Reply:
I still don't understand the MustRestart change.

Reply:
Given https://github.com/scylladb/scylladb/pull/14382, I think we're good with `liveness::LiveUpdate` now.